kqueue.go

```diff
@@ -119,6 +121,13 @@ func (w *Watcher) Remove(name string) error {
 	w.mu.Lock()
 	isDir := w.paths[watchfd].isDir
 	delete(w.watches, name)
+
+	parentName := filepath.Dir(name)
+	delete(w.watchesByDir[parentName], watchfd)
+	if len(w.watchesByDir[parentName]) == 0 {
```

Comment:
Feel free to add more commits instead of amending so its easier for me to tell whats changed between reviews. We can squash at the end.